repo/packages/K/kibana/800/config.json

```diff
@@ -0,0 +1,130 @@
+            "properties": {
+              "key": {
+                "type": "string"
+              },
+              "value": {
+                "type": "string"
+              }
+            },
+            "additionalProperties": false,
+            "default": {}
+          },
+          "default": []
+        }
+      },
+      "required": [
+        "name",
+        "user"
+      ]
+    },
+    "kibana": {
+      "description": "Kibana service configuration properties",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "plugins": {
+          "description": "An optional comma-separated list of URLs to ZIP files containing Kibana plugins.",
+          "type": "string",
+          "default": ""
+        },
+        "elasticsearch_url": {
+          "description": "The URL of the Elasticsearch cluster to use for all your queries.",
+          "type": "string",
+          "default": "http://coordinator.elastic.l4lb.thisdcos.directory:9200"
+        },
+        "elasticsearch_tls": {
+          "description": "Enable TLS encryption for communicating with Elastic (required when Elastic has TLS support enabled).",
+          "type": "boolean",
+          "default": false
+        },
+        "elasticsearch_xpack_security_enabled": {
+          "description": "Set it to true when Elasticsearch X-Pack security is enabled. Defaults to false",
+          "type": "boolean",
+          "default": false
+        },
+        "user": {
+          "description": "Kibana username to use for X-Pack authentication, if enabled",
+          "type": "string",
+          "default": "kibana"
+        },
+        "password": {
+          "description": "Password to use for X-Pack authentication, if enabled. Note that you are not setting the password here. You are specifying the credentials for Kibana to use when sending requests.",
+          "type": "string",
+          "default": "changeme"
+        },
+        "placement": {
+          "type": "array",
+          "items": {
+            "type": "array",
+            "items": [
+              {
+                "type": "string"
+              }
+            ]
+          },
+          "default": []
+        },
+        "cpus": {
+          "description": "Kibana CPU requirements",
+          "type": "number",
+          "default": 0.5
+        },
+        "mem": {
+          "description": "Kibana mem requirements",
+          "type": "integer",
+          "default": 2048
+        },
+        "custom_kibana_yml": {
+          "description": "Custom YAML to be appended to kibana.yml. This field must be base64 encoded.",
+          "type": "string",
+          "media": {
+            "binaryEncoding": "base64",
+            "type": "application/x-yaml"
+          },
+          "default": ""
+        },
+        "xpack_enabled": {
+          "description": "[DEPRECATED in 2.5.0-6.3.2] X-Pack is enabled by default. Changing this setting has no effect.",
```

Comment:
I think the setting deprecation strategy so far has been leaving deprecated fields in the configuration schema file so that SDK-based upgrades (`dcos $package update start`) work. Otherwise, SDK complains about schema validation.

Reply:
If we delete them users would have to do a Marathon-style upgrade. I think even assuming we're releasing a breaking version, we should maintain convenience to users by allowing them to do an SDK-based upgrade.

Reply:
fair point - doesn't harm us to leave it. Was asking more so in the sense that if we're bumping package version to `3.0.0-x` then it signifies breaking changes and we should get them all out of the way. But if this would be the only inconvenience then you're absolutely right let's keep it simple for users.